crates/graphql/src/schema/query_root.rs

```diff
@@ -428,11 +428,41 @@ impl QueryRoot {
         Ok(Wallet::new(address, twitter_handle))
     }
 
-    fn listings(&self, context: &AppContext) -> FieldResult<Vec<Listing>> {
+    async fn listings(
+        &self,
+        context: &AppContext,
+        #[graphql(description = "Return only listings whose NFTs' metadatas contain this term")]
+        term: Option<String>,
+        #[graphql(description = "Query limit")] limit: i32,
+        #[graphql(description = "Query offset")] offset: Option<i32>,
+    ) -> FieldResult<Vec<Listing>> {
         let now = Local::now().naive_utc();
-        let conn = context.shared.db.get()?;
+        let conn = context.shared.db.get().context("Failed to connect to DB")?;
+
+        let nft_metadata_addresses: Option<Vec<String>> = match term {
```

Comment:
this had to be separated from the additional query filter because rust couldnt pass the query through to the async block this creates.